packages/rocketchat-ui/client/lib/msgTyping.js

```diff
@@ -24,6 +31,10 @@ export const MsgTyping = (function() {
 	};
 
 	const addStream = function(room) {
+		if (disableUserTyping === true) {
+			RocketChat.Notifications.unRoom(room, 'typing', function() { return; });
```

Comment:
The third argument must be the original event listener callback attached to the event, e.g.,

```js
// The event listener callback (cb) is attached to the event
const cb = () => { ... };
RocketChat.Notifications.onRoom(room, 'typing', cb);

...

// The same event listener callback is detached
RocketChat.Notifications.unRoom(room, 'typing', cb);
```